backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/switch/BpkSwitch.kt

```diff
@@ -73,15 +79,25 @@ fun BpkSwitch(
     Row(
         verticalAlignment = Alignment.CenterVertically,
         horizontalArrangement = Arrangement.SpaceBetween,
-        modifier = modifier.applyIf(onCheckedChange != null) {
-            toggleable(
-                value = checked,
-                role = Role.Switch,
-                interactionSource = interactionSource,
-                indication = null,
-                onValueChange = onCheckedChange!!,
-            )
-        },
+        modifier = modifier
+            .applyIf(onCheckedChange != null) {
+                toggleable(
+                    value = checked,
+                    role = Role.Switch,
+                    interactionSource = interactionSource,
+                    indication = null,
+                    onValueChange = onCheckedChange!!,
+                    enabled = enabled,
+                )
+            }.applyIf(onCheckedChange == null) {
+                semantics(mergeDescendants = true) {
+                    role = Role.Switch
+                    if (!enabled) {
+                        disabled()
+                        toggleableState = ToggleableState(checked)
+                    }
+                }.focusable()
```

Comment:
i don't think you actually need this

Reply:
without it the screenreader skips over the disabled switches :(

Reply:
hmm odd it doesn't do for me? if you look at the implementation it actually doesn't do anything if the `enabled = false` property is set in the function, so it is not expected to be used for disabled items

Reply:
oop it's my laziness that made me think it wasn't working. i've been using arrow keys to move the screenreader focus as using the touchpad to swipe properly has been really temperamental. 

When swiping properly the screenreader doesn't skip 👍